WordPress/gutenberg
# ComboboxControl and FormTokenField: enhance components with custom render callback for options

## What?
This PR
- Adds an optional render callback to the `SuggestionsList` and `ComboboxControl` components in order to allow the consumer to control how an option is rendered inside `SuggestionsList` for a combobox or any other consumer of `SuggestionsList`
- Adds a corresponding story to `ComboboxControl` to demonstrate the new behavior

## Why?
In many use cases of combobox, we need to show more detail to the user for an option in the list apart from just the label. More details help users identify the correct option from the list when there are plenty of them available with similar labels. 

## How?
- It first adds an optional (thus full backward compatible) prop `renderSuggestion` to `SuggestionsList` component.
- If `renderSuggestion` is passed, it will be given preference and the suggestion will be passed to it to allow it to render the suggestion.
- It also adds `renderOption` prop to `ComboboxControl` which in turn gets passed to `SuggestionsList` above.

## Testing Instructions

- Checkout the branch
- Run `npm run storybook:dev`
- Check combobox story in Storybook
- Confirm that the existing story behaves as before i.e. it is backward compatible
- Confirm that `With Render Option` story shows more details in the options for combobox like Age, Country.

## Screenshots or screencast

<img width="1044" alt="Screenshot 2022-07-21 at 5 49 24 PM" src="https://user-images.githubusercontent.com/18226415/180211982-709e98a0-779e-4adf-9924-7b629c664d3b.png">



---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202011579628273
